core/src/main/java/org/springframework/security/scheduling/DelegatingSecurityContextTaskScheduler.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.scheduling;
+
+import org.springframework.core.task.AsyncTaskExecutor;
+import org.springframework.scheduling.SchedulingTaskExecutor;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.scheduling.Trigger;
+import org.springframework.security.concurrent.DelegatingSecurityContextCallable;
+import org.springframework.security.concurrent.DelegatingSecurityContextRunnable;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.task.DelegatingSecurityContextAsyncTaskExecutor;
+
+import java.util.Date;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledFuture;
+
+/**
+ * An {@link SchedulingTaskExecutor} which wraps each {@link Runnable} in a
+ * {@link DelegatingSecurityContextRunnable} and each {@link Callable} in a
+ * {@link DelegatingSecurityContextCallable}.
+ *
+ * @author Richard Valdivieso
+ * @since 5.1
+ *
+ */
+public class DelegatingSecurityContextTaskScheduler extends
+		DelegatingSecurityContextAsyncTaskExecutor implements TaskScheduler {
```

Comment:
We should not extend `DelegatingSecurityContextAsyncTaskExecutor` since `TaskExecutor` is not a `TaskExecutor`. Instead the class can use it's own wrap method directly.